tests/IBTK/rt0_coarsen_01.cpp

```diff
@@ -0,0 +1,194 @@
+    SAMRAIManager::startup();
+
+    // prevent a warning about timer initializations
+    TimerManager::createManager(nullptr);
+    {
+        Pointer<AppInitializer> app_initializer = new AppInitializer(argc, argv, "rt0.log");
+        Pointer<Database> input_db = app_initializer->getInputDatabase();
+
+        // Create major algorithm and data objects that comprise the
+        // application.
+        Pointer<CartesianGridGeometry<NDIM> > grid_geometry = new CartesianGridGeometry<NDIM>(
+            "CartesianGeometry", app_initializer->getComponentDatabase("CartesianGeometry"));
+        Pointer<PatchHierarchy<NDIM> > patch_hierarchy = new PatchHierarchy<NDIM>("PatchHierarchy", grid_geometry);
+        Pointer<StandardTagAndInitialize<NDIM> > error_detector = new StandardTagAndInitialize<NDIM>(
+            "StandardTagAndInitialize", NULL, app_initializer->getComponentDatabase("StandardTagAndInitialize"));
+        Pointer<BergerRigoutsos<NDIM> > box_generator = new BergerRigoutsos<NDIM>();
+        Pointer<LoadBalancer<NDIM> > load_balancer =
+            new LoadBalancer<NDIM>("LoadBalancer", app_initializer->getComponentDatabase("LoadBalancer"));
+        Pointer<GriddingAlgorithm<NDIM> > gridding_algorithm =
+            new GriddingAlgorithm<NDIM>("GriddingAlgorithm",
+                                        app_initializer->getComponentDatabase("GriddingAlgorithm"),
+                                        error_detector,
+                                        box_generator,
+                                        load_balancer);
+
+        Pointer<VisItDataWriter<NDIM> > visit_data_writer = app_initializer->getVisItDataWriter();
+
+        // Create variables and register them with the variable database.
+        VariableDatabase<NDIM>* var_db = VariableDatabase<NDIM>::getDatabase();
+        Pointer<VariableContext> ctx = var_db->getContext("context");
+        Pointer<SideVariable<NDIM, double> > u_sc_var = new SideVariable<NDIM, double>("u_sc");
+        const int u_sc_idx = var_db->registerVariableAndContext(u_sc_var, ctx, IntVector<NDIM>(4));
+        Pointer<SideVariable<NDIM, double> > exact_sc_var = new SideVariable<NDIM, double>("exact_sc");
+        const int exact_sc_idx = var_db->registerVariableAndContext(exact_sc_var, ctx, IntVector<NDIM>(4));
+        // TODO u_cc_var is only for plotting (and testing): remove later
+#define DO_PLOT
+#ifdef DO_PLOT
+        Pointer<CellVariable<NDIM, double> > u_cc_var = new CellVariable<NDIM, double>("u_cc", NDIM);
+        const int u_cc_idx = var_db->registerVariableAndContext(u_cc_var, ctx);
+        Pointer<CellVariable<NDIM, double> > exact_cc_var = new CellVariable<NDIM, double>("exact_cc", NDIM);
+        const int exact_cc_idx = var_db->registerVariableAndContext(exact_cc_var, ctx);
+#endif
+
+        gridding_algorithm->makeCoarsestLevel(patch_hierarchy, 0.0);
+        const int tag_buffer = std::numeric_limits<int>::max();
```

Comment:
I think that setting this `numeric_limits<int>::max()` will effectively refine the entire domain --- is that the goal?

You also can specify the region to be refined in the input file using `REFINED_BOXES`. There are some examples of this in the repo.

Reply:
The goal is to refine the entire domain (I want a domain with two patches, one on each level). I will look for refined boxes.